cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -386,10 +386,16 @@ func (o *Options) ApplyTo(config *apiserver.Config) error {
 	gardenerAPIServerConfig := config.GenericConfig
 
 	gardenerVersion := version.Get()
+	gardenerAPIServerConfig.Version = &gardenerVersion
 	gardenerAPIServerConfig.OpenAPIV3Config = genericapiserver.DefaultOpenAPIV3Config(openapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(api.Scheme))
 	gardenerAPIServerConfig.OpenAPIV3Config.Info.Title = "Gardener"
 	gardenerAPIServerConfig.OpenAPIV3Config.Info.Version = gardenerVersion.GitVersion
-	gardenerAPIServerConfig.Version = &gardenerVersion
+
+	// For backwards-compatibility, we also have to keep serving the /openapi/v2 endpoint since kubectl < 1.27 rely on
```

Comment:
Maybe add a todo to drop it once we drop support of Kubernetes version 1.27
```suggestion
	// For backward-compatibility, we also have to keep serving the /openapi/v2 endpoint since kubectl < 1.27 rely on
```

Reply:
I thought about adding a TODO, though I'm not sure when would be the best time to drop it (even if `1.27` is the lowest supported version, people should probably still be able to use `kubectl < 1.27`). I guess we have to drop it once the configuration struct no longer supports it, and can keep it until this happens.